MTA-6893 Document Hub OIDC authentication (#382)
migtools/mta-documentation: docs/topics/mta-install/proc_generate-api-key-web-console.adoc

:_newdoc-version: 2.18.5
:_template-generated: 2025-07-31
:_mod-docs-content-type: PROCEDURE

[id="generate-api-key-web-console_{context}"]
= Generate API key from the web console

[role="_abstract"]

> You can generate application programming interface (API) keys in the {ProductFirstRef} web console. The API key inherits permissions from the authenticated user's roles and has a maximum lifespan limit of 10 years. You can optionally configure the lifespan period when you create a token.

Reviewer: ```suggestion
As an administrator, you can generate application programming interface (API) keys in the {ProductFirstRef} web console. An API key inherits permissions from the authenticated user's roles and has a maximum lifespan of 10 years. You can configure the lifespan when you create a token.
```

Change the first sentence to start with "As an administrator, you can..." to maintain the consistent conversational tone established in previous reviews.

"maximum lifespan limit" is redundant. Change to "maximum lifespan".
"lifespan period" is also redundant. Change to "lifespan".

As noted in previous reviews, "optionally" is redundant when paired with the auxiliary verb "can". Simply say "You can configure the lifespan..."

Author: I had to split this introduction to keep it under 300 characters. Thanks for the review!

.Prerequisites

* You installed {ProductShortName} 8.2.0 in a Red Hat OpenShift cluster.
* You are an {ProductShortName} Administrator.

> .Procedure
> 
> . Log in to the {ProductShortName} application.
> . In the Administration mode, go to *User Management*.
> . Select *Tokens*.
> . Click *Create API Key*.
> . Optionally enter the lifespan of the token.
> . Click *Create*.
> . Copy the displayed token.
> +
> You can use the API key for the {mta-dl-full} agent.

Reviewer: ```suggestion
.Procedure

. Log in to the {ProductShortName} application.
. In the *Administration* mode, click *User Management*.
. Click *Tokens*.
. Click *Create API Key*.
. Optional: Enter the lifespan of the token.
. Click *Create*.
. Copy the displayed token.
+
You can use the API key for the {mta-dl-full} agent.
```

User Focus: Shift from third-person perspectives to the active second-person (e.g., "As an administrator, you can...") to foster a more engaging, conversational tone
.
Minimalism: Eliminate redundant phrases and filler words to keep sentences concise. For example, "maximum lifespan limit" was shortened to "maximum lifespan," and "optionally" was removed when already paired with the auxiliary verb "can."

Articles: Do not omit articles to save space; include the indefinite article "an" before abbreviations that start with a vowel sound, such as "an API key". 

Capitalization: Use lowercase for common nouns and generic personas, such as "administrator," when they appear in the middle of a sentence.

Red Hat Branding: Specify the full product distribution on its first occurrence, such as expanding "OpenShift" to "Red Hat OpenShift Container Platform".

Non-breaking Spaces: Always use a non-breaking space ({nbsp}) between the words "Red" and "Hat" to prevent the company name from splitting across a line break.

Prepositions: Use the preposition "on" rather than "in" when describing software installed on a cluster.

Prerequisites: Write prerequisite items as checks that the user must have completed prior to the task, utilizing the present perfect tense (e.g., "You have installed...") to maintain grammatical parallelism.

UI Highlighting: Enclose all user interface controls
